docs/assertions.md

```diff
@@ -856,3 +856,20 @@ function test_failure() {
 }
 ```
 :::
+
+## fail
+> `fail "failure message"`
+
+Unambiguously reports an error message. Useful for reporting specific message
+when testing situations not covered by any `assert_*` functions.
+
+::: code-group
+```bash [Example]
+function test_success() {
+  true || fail "This will never fail"
+}
+function test_failure() {
+  [ $(date +%-H) -lt 6 ] || fail "It's late, go to bed"
+}
```

Comment:
The example you set for test_failure doesn't always fail because it depends on the time of execution. The idea is to have one of the examples always succeed and the other always fail. Could you change it?

I think it would also help to use the if syntax instead of the || operator in both examples, as it's more familiar to a greater number of programmers.

Reply:
I've modified them as requested. I had trouble coming up with an example that could seem useful in only a line or two, but I think this gets the job done.